### PR TITLE
feat: data platform infra docs - 6361

### PR DESCRIPTION
**Pull request by motylp**
#### 📲 What

Data Platform infrastructure documentation and documentation formatting cleanup.

Ticket: [6361](https://amido-dev.visualstudio.com/Amido-Stacks/_workitems/edit/6361)

#### 🤔 Why
		
Data Platform documentation.
		
#### 🛠 How
		
MD files and sidebards.js updated.

#### 👀 Evidence
		
See below.
		 
#### 🕵️ How to test

You can render the documentation locally using `npm`:
```
npm init
npm install
npm run build
npm run serve
```

#### ✅ Acceptance criteria Checklist

- [x] Code peer reviewed?
- [x] Documentation has been updated to reflect the changes?
- [x] Passing all automated tests, including a successful deployment?
- [x] Passing any exploratory testing?
- [x] Rebased/merged with latest changes from development and re-tested?
- [x] Meeting the Coding Standards?

### CODE REVIEW

File: docs/workloads/azure/data/components/infrastructure/core_data_platform_azure.md [hunk 1]
```diff
@@ -10,28 +10,84 @@ keywords:
...
+    * Service Principal Secret
+    * Azure Tenant ID – Directory ID for Azure Active Directory application
+    * Azure Client ID – Application ID for Azure Active Directory application
+    * Names of the ADLS and Blob storage accounts
+    * Other secret names with empty values to be replaced manually. Existing secrets are not
+    overwritten.
+5. **Azure Data Lake Storage Gen2**
+6. **Azure Blob Storage**
+7. **Databricks Workspace** including:
+    * **Key Vault-backed secret scope**
```
Comment (Trishisingh) [inline docs/workloads/azure/data/components/infrastructure/core_data_platform_azure.md:47]: It is better to add a few more details around secure data brick cluster connectivity as it adds few subnets nsg, etc.

File: docs/workloads/azure/data/components/infrastructure/core_data_platform_azure.md [hunk 2]
```diff
@@ -10,28 +10,84 @@ keywords:
...
-    * Azure Key Vault-backed secret scope
+`infra` subfolder contains the following definitions:
 
+1. **Resource Group**
+2. **MS SQL Database** instance with database schemas
+3. **Key Vault**
+4. **Key Vault Secrets**:
+    * Connection strings for the created SQL databases
+    * Passwords to the SQL Databases
+    * Service Principal Secret
```
Comment (Trishisingh) [inline docs/workloads/azure/data/components/infrastructure/core_data_platform_azure.md:38]: I don't think we have Service Principal Secret ?

File: docs/workloads/azure/data/components/infrastructure/core_data_platform_azure.md [hunk 3]
```diff
@@ -10,28 +10,84 @@ keywords:
...
+    * Azure Tenant ID – Directory ID for Azure Active Directory application
+    * Azure Client ID – Application ID for Azure Active Directory application
```
Comment (Trishisingh) [inline docs/workloads/azure/data/components/infrastructure/core_data_platform_azure.md:40]: Also no clientID

File: docs/workloads/azure/data/components/infrastructure/core_data_platform_azure.md (hunk 1 above)
Comment (motylp) [reply docs/workloads/azure/data/components/infrastructure/core_data_platform_azure.md:47]: OK, as agreed, I will add it in a separate PR, as this is a fresh code and I didn't have much chance to get to know it.

**Pull request merged**